fairing/preprocessors/function.py

```diff
@@ -36,6 +40,9 @@ def __init__(self,
             path_prefix=path_prefix,
             input_files=input_files)
 
+        if notebook_util.is_in_notebook():
+            logger.warning("The FunctionPreProcessor is typically used for notebook environment.")
```

Comment:
Adding little more detail would be helpful. "The Function PreProcessor is optimized for using in a notebook or IPython environment. For it to work, the python version should be same for both local python and the python in the docker. Please look at alternatives like BasePreprocessor or FullNotebookPreprocessor."

Being more verbose with error/warn messages is always a good thing.

Reply:
Great warning message!